crates/bevy_tasks/src/static_task_pool.rs

```diff
@@ -0,0 +1,576 @@
+        }
+    }
+
+    /// The number of threads active in the task pool.
+    pub fn thread_num(&self) -> usize {
+        self.thread_count.load(Ordering::Relaxed)
+    }
+
+    /// Checks if the threads for the task pool have been started.
+    pub fn is_initialized(&self) -> bool {
+        !self.threads.lock().unwrap().is_empty()
+    }
+
+    /// Initializes the task pool with the configuration in
+    ///
+    /// # Panics
+    /// Panics if the task pool was already initialized or provided a builder that
+    /// yields zero threads.
+    pub fn init(
+        &'static self,
+        builder: TaskPoolBuilder,
+    ) -> Result<(), TaskPoolInitializationError> {
+        let mut join_handles = self.threads.lock().unwrap();
+
+        if !join_handles.is_empty() {
+            // TODO: figure out a way to support reconfiguring/reinitializing StaticTaskPools.
+            return Err(TaskPoolInitializationError::AlreadyInitialized);
+        }
+
+        let num_threads = builder
+            .num_threads
+            .unwrap_or_else(crate::available_parallelism);
+
+        if num_threads == 0 {
+            return Err(TaskPoolInitializationError::ZeroThreads);
+        }
+
+        *join_handles = Vec::with_capacity(num_threads);
+        for i in 0..num_threads {
+            let thread_name = if let Some(thread_name) = builder.thread_name.as_deref() {
+                format!("{thread_name} ({i})")
+            } else {
+                format!("TaskPool ({i})")
+            };
+            let mut thread_builder = std::thread::Builder::new().name(thread_name);
+
+            if let Some(stack_size) = builder.stack_size {
+                thread_builder = thread_builder.stack_size(stack_size);
+            }
+
+            let on_thread_spawn = builder.on_thread_spawn.clone();
+
+            let res = thread_builder.spawn(move || {
+                TaskPool::LOCAL_EXECUTOR.with(|local_executor| {
+                    if let Some(on_thread_spawn) = on_thread_spawn {
+                        on_thread_spawn();
+                        drop(on_thread_spawn);
+                    }
+                    loop {
+                        let res = std::panic::catch_unwind(|| {
+                            let tick_forever = async move {
+                                loop {
+                                    local_executor.tick().await;
+                                }
+                            };
+                            block_on(self.executor.run(tick_forever));
+                        });
+                        if res.is_ok() {
+                            break;
+                        }
+                    }
+                });
+            });
+            match res {
+                Ok(join_handle) => {
+                    join_handles.push(join_handle);
+                }
+                Err(join_handle) => {
+                    *join_handles = Vec::new();
+                    drop(join_handles);
```

Comment:
Why explicitly drop the `guard` here?

Reply:
This used to be a panic, which would poison the lock. I'll remove this.

Reply:
Done.